crypto/src/main/java/org/springframework/security/crypto/password/AbstractValidatingPasswordEncoder.java

```diff
@@ -18,6 +18,14 @@
 
 import org.jspecify.annotations.Nullable;
 
+import org.springframework.util.StringUtils;
+
+/**
+ * Implementation of PasswordEncoder.
```

Comment:
I wouldn't quite say that this is an implementation as much as it is an abstract class that allows subclasses to except the password to be non-{@code null}.

Perhaps change this to:

```java
/**
  * An abstract {@link PasswordEncoder} that implementers can use for expecting the
  * password to be non-{@code null}. Each common password API method is accompanied
  * with an abstract method with a {@code NonNull} prefix. By implementing this, the concrete
  * class is specifying what to do with the password when it is non-{@code null}, allowing this
  * class to handle the {@code null} case.
  * 
  * @author Rob Winch
  * @since 7.0
  */
```